src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.cpp

```diff
@@ -203,9 +201,18 @@ bool ChaseMovementGenerator::Update(Unit* owner, uint32 diff)
 
             owner->AddUnitState(UNIT_STATE_CHASE_MOVE);
 
+            bool walk = false;
+            if (cOwner)
+            {
+                if (cOwner->GetMovementTemplate().CanChaseWalking() && owner->IsWalking())
```

Comment:
switch on Chase field would look nicer imo
You need to ignore it for pets (any player controlled? needs verification)

Reply:
Agree on switch. Ignores pets now.